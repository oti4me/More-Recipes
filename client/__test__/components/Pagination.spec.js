import React from 'react';
import ConnectedPagination, {
  Pagination
} from '../../src/components/Pagination';

describe('Routes', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Pagination />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const wrapper = shallow(<ConnectedPagination store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});