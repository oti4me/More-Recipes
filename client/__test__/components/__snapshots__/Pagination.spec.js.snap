// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <PaginationBoxView
        activeClassName="active"
        breakClassName="break-me"
        breakLabel={
          <Link
            replace={false}
            to="!#"
          >
            ...
          </Link>
        }
        containerClassName="pagination"
        disableInitialCallback={false}
        disabledClassName="disabled"
        marginPagesDisplayed={5}
        nextClassName="next"
        nextLabel="next"
        onPageChange={[Function]}
        pageCount={1}
        pageRangeDisplayed={5}
        previousClassName="previous"
        previousLabel="previous"
        subContainerClassName="pages pagination"
      />,
      "className": "pagination",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeClassName": "active",
        "breakClassName": "break-me",
        "breakLabel": <Link
          replace={false}
          to="!#"
        >
          ...
        </Link>,
        "containerClassName": "pagination",
        "disableInitialCallback": false,
        "disabledClassName": "disabled",
        "marginPagesDisplayed": 5,
        "nextClassName": "next",
        "nextLabel": "next",
        "onPageChange": [Function],
        "pageCount": 1,
        "pageRangeDisplayed": 5,
        "previousClassName": "previous",
        "previousLabel": "previous",
        "subContainerClassName": "pages pagination",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <PaginationBoxView
          activeClassName="active"
          breakClassName="break-me"
          breakLabel={
            <Link
              replace={false}
              to="!#"
            >
              ...
            </Link>
          }
          containerClassName="pagination"
          disableInitialCallback={false}
          disabledClassName="disabled"
          marginPagesDisplayed={5}
          nextClassName="next"
          nextLabel="next"
          onPageChange={[Function]}
          pageCount={1}
          pageRangeDisplayed={5}
          previousClassName="previous"
          previousLabel="previous"
          subContainerClassName="pages pagination"
        />,
        "className": "pagination",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeClassName": "active",
          "breakClassName": "break-me",
          "breakLabel": <Link
            replace={false}
            to="!#"
          >
            ...
          </Link>,
          "containerClassName": "pagination",
          "disableInitialCallback": false,
          "disabledClassName": "disabled",
          "marginPagesDisplayed": 5,
          "nextClassName": "next",
          "nextLabel": "next",
          "onPageChange": [Function],
          "pageCount": 1,
          "pageRangeDisplayed": 5,
          "previousClassName": "previous",
          "previousLabel": "previous",
          "subContainerClassName": "pages pagination",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Routes should render correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Pagination)
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "getAllRecipes": [Function],
      "getFavourites": [Function],
      "getMyRecipes": [Function],
      "pagination": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "user": Object {},
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "getAllRecipes": [Function],
        "getFavourites": [Function],
        "getMyRecipes": [Function],
        "pagination": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "user": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
